Content.Server/AI/Operators/Generic/BarkOperator.cs

```diff
@@ -0,0 +1,42 @@
+using Content.Server.AI.Utility.AiLogic;
+using Content.Server.GameObjects.Components.Movement;
+using Robust.Shared.Interfaces.GameObjects;
+
+namespace Content.Server.AI.Operators.Generic
+{
+    public class BarkOperator : AiOperator
+    {
+        private IEntity _owner;
+        private BarkType _barkType;
+
+        public BarkOperator(IEntity owner, BarkType barkType)
+        {
+            _owner = owner;
+            _barkType = barkType;
+        }
+        
+        public override Outcome Execute(float frameTime)
+        {
+            if (!_owner.TryGetComponent(out AiControllerComponent aiController))
+            {
+                return Outcome.Failed;
+            }
+
+            if (aiController.Processor is UtilityAi utilityAi)
+            {
+                utilityAi.Bark(_barkType);
+            }
+            
+            return Outcome.Success;
+        }
+    }
+
+    /// <summary>
+    /// Each AI might respond to a bark differently so we'll pass in
+    /// the action type (e.g. Reloading) and then they can respond in their own way.
+    /// </summary>
+    public enum BarkType
+    {
+        
+    }
```

Comment:
Is this meant to be empty?

Reply:
I still wasn't entirely sure how to tackle barks at that point but I'll delete it as it's not the method I was going to try, cheers :)